database/service.py

```diff
@@ -0,0 +1,256 @@
+        }
+
+    def get_ongoing_vote(self, proposal_id: str) -> Optional[Dict[str, Any]]:
+        """Get an ongoing vote by proposal ID"""
+        logger.info("Retrieving ongoing vote with proposal_id: %s", proposal_id)
+        session = self._get_session()
+        vote = session.query(OngoingVote).filter_by(proposal_id=proposal_id).first()
+        if self._session is None:
+            session.close()
+        if not vote:
+            logger.info("No ongoing vote found with proposal_id: %s", proposal_id)
+            return None
+        logger.info("Found ongoing vote: %s", vote.title)
+        return {
+            "proposal_id": vote.proposal_id,
+            "draft": vote.draft,
+            "end_time": vote.end_time,
+            "title": vote.title,
+            "channel_id": vote.channel_id,
+            "thread_id": vote.thread_id,
+            "message_id": vote.message_id,
+        }
+
+    def save_ongoing_vote(self, vote_data: Dict[str, Any]) -> None:
+        """Save or update an ongoing vote"""
+        logger.info(
+            "Saving ongoing vote with proposal_id: %s", vote_data.get("proposal_id")
+        )
+        session = self._get_session()
+        try:
+            vote = (
+                session.query(OngoingVote)
+                .filter_by(proposal_id=vote_data["proposal_id"])
+                .first()
+            )
+            if vote:
+                logger.info("Updating existing ongoing vote")
+                for key, value in vote_data.items():
+                    setattr(vote, key, value)
+            else:
+                logger.info("Creating new ongoing vote")
+                vote = OngoingVote(**vote_data)
+                session.add(vote)
+            session.commit()
+            logger.info("Successfully saved ongoing vote")
+        except Exception as e:
+            logger.error("Error saving ongoing vote: %s", str(e))
+            session.rollback()
+            raise
+        finally:
+            if self._session is None:
+                session.close()
+
+    def get_posted_events(self) -> List[int]:
+        """Get list of posted event IDs"""
+        logger.info("Retrieving all posted events")
+        session = self._get_session()
+        events = session.query(Event).filter(Event.posted_at.isnot(None)).all()
+        if self._session is None:
+            session.close()
+        event_ids = [int(event.event_id) for event in events]
+        logger.info("Found %d posted events", len(event_ids))
+        return event_ids
+
+    def save_posted_event(self, event_id: int, guild_id: int):
+        """Save a posted event"""
+        session = self._get_session()
+        event = Event(
+            event_id=event_id, guild_id=guild_id, posted_at=datetime.now().timestamp()
+        )
+        session.add(event)
+        session.commit()
+        if self._session is None:
+            session.close()
+
+    def get_notified_events(self) -> Dict[int, float]:
```

Comment:
might be runtime issues here in that the model has `notified_at` as a "BigInteger" type now